rust/src/utils/dev.rs

```diff
@@ -0,0 +1,87 @@
+// TODO: Would be clearer with cfg_if
+
+#[cfg(target_os = "freebsd")]
+pub use freebsd::*;
+#[cfg(target_os = "linux")]
+pub use linux::*;
+#[cfg(target_os = "solaris")]
```

Comment:
Why bother with Solaris?  If anything, you should use `illumos`.  In Rust, those are considered separate operating systems.